src/cargo/core/resolver/conflict_cache.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::{HashMap, HashSet};
+
+use core::{Dependency, PackageId};
+use core::resolver::{ConflictReason, Context};
+
+pub(super) struct ConflictCache {
```

Comment:
I think this `pub(super)` can change to `pub` like the ones below, right?

Reply:
If I remove it from the type, I get:
```private type `core::resolver::ConflictReason` in public interface``` on the methods.
I could just make everything `pub`, but these are all just implementation details of `core::resolver` so I think `pub(super)` is appropriate.